simulator.js

```diff
@@ -37,6 +42,14 @@ function execute( environment, text_input ) {
     handle_command( environment, command );
 }
 
+// executes the next instruction pointed by the program counter
+// this fn is to be used only if instructions have been loaded into memory
+function step( environment ){
```

Comment:
Could you add a test for this function?

Reply:
Ok. I will work on it. First I need to make sure I learn Mocha/Chai a littel before I give it a try. Testing the step function is going to be more challenging because the output depends on what is in the memory and what instruction is being executed. I did a preliminary test of the step function by running the console-simulator.js with real instructions stored in memory. So it is tested, but not thoroughly and not with an automated test.

Reply:
You can create a super simple program, manually put it in memory, and run it. Then make sure the resulting memory & registers equals what is expected. It is a bit of work.

Reply:
I think a one line program should be enough as an initial test.

Reply:
Great, I like the new test. Feel free to merge when your ready.